gems/smithy/lib/smithy/welds.rb

```diff
@@ -1,18 +1,18 @@
 # frozen_string_literal: true
 
 require_relative 'welds/endpoints'
+require_relative 'welds/plugins'
 require_relative 'welds/rubocop'
 
 module Smithy
   # @api private
   module Welds
     @welds = {}
 
+    # Weld.subclasses should be ordered by insertion order. Reverse the order
```

Comment:
How is insertion order controlled?

Reply:
By when the class is defined. You should verify this. ObjectSpace had no such guarantee.

Reply:
Haven't done a deep dive on Active Support's DescendantsTracker is that there are no specific guarantees, particularly across different ruby implementations (jruby, truffle, ect).

Reply:
This is not from active support, this is native ruby https://ruby-doc.org/core-3.1.0/Class.html#method-i-subclasses

Reply:
Ah fair - but yeah either way, order is not guaranteed.   Long term I think we want/need some ability to control priority.  There are a lot of ways we can do that: 1) `priority` field (as we did for RubyIntegrations and we do still for handlers) 2) some sort of configuration that allows order to be specified.  3) Some sort of relative ordering (through configuration or specified as an attirbute of a given weld).

But we don't need to figure that out now I think.

Reply:
Insertion order may not even be correct either because it can depend on whether you define the weld before or after you require smithy/welds.